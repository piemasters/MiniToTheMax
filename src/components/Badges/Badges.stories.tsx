import type { Meta, StoryObj } from '@storybook/react-vite';

import { Badges } from './Badges';

const meta: Meta<typeof Badges> = {
  component: Badges,
};

export default meta;
type Story = StoryObj<typeof Badges>;

export const Primary: Story = {
  render: () => <Badges />,
};
