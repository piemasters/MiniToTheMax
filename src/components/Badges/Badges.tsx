import React, { FC } from 'react';
import kebabCase from 'lodash.kebabcase';

import { Badge } from '../Badge';
import { PageLink } from '../PageLink';
import { BadgeNames } from '../../types';

/**
 * The Badges component is collection of badges with the most popular categories to display on the homepage.
 */
export const Badges: FC = () => {
  const badges: BadgeNames[] = [
    BadgeNames.showcase,
    BadgeNames.tutorials,
    BadgeNames.reviews,
    BadgeNames.battleReports,
    //BadgeNames.build,
  ];

  return (
    <div
      className="flex flex-row flex-wrap justify-center"
      data-testid="badges"
    >
      {badges.map((badge: BadgeNames) => {
        return (
          <PageLink
            to={kebabCase(badge)}
            key={badge}
            className="flex flex-col no-underline opacity-90 hover:opacity-100"
            type={'cover'}
            direction={'up'}
          >
            <Badge type={badge} />
          </PageLink>
        );
      })}
    </div>
  );
};

export default Badges;
