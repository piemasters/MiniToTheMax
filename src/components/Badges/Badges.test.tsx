import { getByTestId, render } from '@testing-library/react';

import { Badges } from './Badges';

describe('Badges', () => {
  test('renders correctly', () => {
    const { container } = render(<Badges />);

    expect(getByTestId(container as HTMLElement, 'badges')).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
