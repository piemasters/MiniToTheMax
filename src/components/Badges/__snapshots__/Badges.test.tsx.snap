// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badges renders correctly 1`] = `
<div>
  <div
    class="flex flex-row flex-wrap justify-center"
    data-testid="badges"
  >
    <a
      class="flex flex-col no-underline opacity-90 hover:opacity-100"
      data-testid="page-link"
      href="/showcase"
    >
      <div
        class="flex flex-col justify-end w-36"
        data-testid="badge"
      >
        <img
          alt=""
          class="w-9/12 opacity-90 mx-auto mb-2 hover:opacity-100"
          src="test-file-stub"
        />
        <p
          class="text-center capitalize"
        >
          showcase
        </p>
      </div>
    </a>
    <a
      class="flex flex-col no-underline opacity-90 hover:opacity-100"
      data-testid="page-link"
      href="/tutorials"
    >
      <div
        class="flex flex-col justify-end w-36"
        data-testid="badge"
      >
        <img
          alt=""
          class="w-9/12 opacity-90 mx-auto mb-2 hover:opacity-100"
          src="test-file-stub"
        />
        <p
          class="text-center capitalize"
        >
          tutorials
        </p>
      </div>
    </a>
    <a
      class="flex flex-col no-underline opacity-90 hover:opacity-100"
      data-testid="page-link"
      href="/reviews"
    >
      <div
        class="flex flex-col justify-end w-36"
        data-testid="badge"
      >
        <img
          alt=""
          class="w-9/12 opacity-90 mx-auto mb-2 hover:opacity-100"
          src="test-file-stub"
        />
        <p
          class="text-center capitalize"
        >
          reviews
        </p>
      </div>
    </a>
    <a
      class="flex flex-col no-underline opacity-90 hover:opacity-100"
      data-testid="page-link"
      href="/battle-reports"
    >
      <div
        class="flex flex-col justify-end w-36"
        data-testid="badge"
      >
        <img
          alt=""
          class="w-9/12 opacity-90 mx-auto mb-2 hover:opacity-100"
          src="test-file-stub"
        />
        <p
          class="text-center capitalize"
        >
          battle Reports
        </p>
      </div>
    </a>
  </div>
</div>
`;
